presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraIntegrationSmokeTest.java

```diff
@@ -428,6 +428,38 @@ public void testUnsupportedColumnType()
         session.execute("DROP KEYSPACE keyspace_6");
     }
 
+    public void testNestedCollectionType()
+    {
+        session.execute("CREATE KEYSPACE keyspace_7 WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_7")
+                .build(), new Duration(1, MINUTES));
+
+        session.execute("CREATE TABLE keyspace_7.table_set (column_5 bigint PRIMARY KEY, nested_collection frozen<set<set<bigint>>>)");
+        session.execute("CREATE TABLE keyspace_7.table_list (column_5 bigint PRIMARY KEY, nested_collection frozen<list<list<bigint>>>)");
+        session.execute("CREATE TABLE keyspace_7.table_map (column_5 bigint PRIMARY KEY, nested_collection frozen<map<int, map<bigint, bigint>>>)");
+
+        assertContainsEventually(() -> execute("SHOW TABLES FROM cassandra.keyspace_7"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("table_set")
+                .row("table_list")
+                .row("table_map")
+                .build(), new Duration(1, MINUTES));
+
+        session.execute("INSERT INTO keyspace_7.table_set (column_5, nested_collection) VALUES (1, {{1, 2, 3}})");
+        assertEquals(execute("SELECT nested_collection FROM cassandra.keyspace_7.table_set").getMaterializedRows().get(0),
+                     new MaterializedRow(DEFAULT_PRECISION, "[[1,2,3]]"));
```

Comment:
How is the set of sets represented in Presto? ie what's `typeof(nested_collection)`?
is it `array(array(bigint))` ?

same question for `list` and `map` in cassandra

Reply:
`typeof(nested_collection)` returns varchar. Current cassandra connector maps varchar for all collection types (list, set and map), so nested collections are also the same.

Reply:
https://github.com/prestosql/presto/issues/660